### PR TITLE
Creating reusable address from/to component.

### PR DESCRIPTION
**Pull request by jpuri**
Ref: #4622

The PR extracts out a re-usable component to display from/to addresses on send screen, this component will be re-used in following [PR](https://github.com/MetaMask/metamask-mobile/pull/5887):

Screenshot 2023-03-06 at 10 28 21 PM

---------------------------------------
QA Steps:
From / To address section in send screen should continue to work as before for various combinations of addresses / tokens.

### CODE REVIEW

**Review by blackdevelopa — APPROVED** [review]

File: app/components/UI/AccountFromToInfoCard/AccountFromToInfoCard.tsx
Comment (digiwand) [inline app/components/UI/AccountFromToInfoCard/AccountFromToInfoCard.tsx:1]: how are these passing the lint check? Are we excluding the indent rule from our ci/lint? 🤔 

Screen Shot 2023-03-13 at 5 41 12 PM

File: app/components/Views/SendFlow/Confirm/index.js [hunk 1]
```diff
@@ -508,77 +471,43 @@ class Confirm extends PureComponent {
...
-      const collectibleTransferInformation =
-        selectedAsset.address.toLowerCase() in
-          collectiblesTransferInformation &&
-        collectiblesTransferInformation[selectedAsset.address.toLowerCase()];
-      if (
-        !collectibleTransferInformation ||
-        (collectibleTransferInformation.tradable &&
-          collectibleTransferInformation.method === 'transferFrom')
-      ) {
-        [, transactionTo] = decodeTransferData('transferFrom', data);
```
Comment (digiwand) [inline app/components/Views/SendFlow/Confirm/index.js:556]: Q: how did we remove this logic to support the collectibles logic and `transferFrom` type? 

I was looking at AccountFromToInfoCard.tsx for a bit and I'm having trouble seeing the full picture

File: app/components/UI/AccountFromToInfoCard/AccountFromToInfoCard.tsx [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+  useEffect(() => {
+    const accountNames =
+      (identities &&
+        Object.keys(identities).map((hash) => identities[hash].name)) ||
+      [];
+    const isOwnAccount = ensRecipient && accountNames.includes(ensRecipient);
+    if (ensRecipient && !isOwnAccount) {
+      setConfusableCollection(collectConfusables(ensRecipient));
+    }
+  }, [identities, ensRecipient]);
```
[the remark this answers is not in the document]
Comment (jpuri) [reply app/components/UI/AccountFromToInfoCard/AccountFromToInfoCard.tsx:97]: Hey @digiwand ; the probabilities of the code running multiple time is too low, but it is logical and correct to add all dependencies in array.

File: app/components/Views/SendFlow/Confirm/index.js (hunk 1 above)
Comment (jpuri) [reply app/components/Views/SendFlow/Confirm/index.js:556]: Interesting thing is this file no longer exists
`import collectiblesTransferInformation from '../../../../util/collectibles-transfer';` but somehow old code was not complaining, as I wrote typescript code it was complained about this and thus I removed it.

File: app/components/UI/AccountFromToInfoCard/AccountFromToInfoCard.tsx
Comment (jpuri) [reply app/components/UI/AccountFromToInfoCard/AccountFromToInfoCard.tsx:1]: Not sure these does not complaint for me. I also tried running `yarn run lint` locally.

**Review by digiwand — CHANGES_REQUESTED** [review]

File: app/components/UI/AccountFromToInfoCard/AccountFromToInfoCard.tsx [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+      setFromAccountName(transactionFromName);
+      return;
+    }
+    (async () => {
+      if (!fromAddress) {
+        return;
+      }
+      const { name: fromName } = identities[fromAddress];
+      const fromEns = await doENSReverseLookup(fromAddress);
+      setFromAccountName(fromEns || fromName);
```
[the remark this answers is not in the document]
Comment (jpuri) [reply app/components/UI/AccountFromToInfoCard/AccountFromToInfoCard.tsx:69]: Hey @digiwand : it was not there in confirm page even before.

[the remark this answers is not in the document]
Comment (digiwand) [reply app/components/UI/AccountFromToInfoCard/AccountFromToInfoCard.tsx:69]: oops, you are right

**Review by digiwand — APPROVED** [review]
LGTM! nice cleanup

File: app/components/Views/SendFlow/Confirm/index.js (hunk 1 above)
Comment (digiwand) [reply app/components/Views/SendFlow/Confirm/index.js:556]: that is interesting

**Comment by seaona** [comment]
Most of the QA happened [in this PR](https://github.com/MetaMask/metamask-mobile/pull/5887) as it integrates both changes. I've reported the issues there.

As per this PR specifically, I have one question. Do we expect the recipient address to be clickable? This allows the ability of adding the address to the address book.  @bschorchit  @jpuri 

It is done already on [Verify Contract details](https://github.com/MetaMask/MetaMask-planning/issues/179) in Mobile and it is also done on Extension in regular Sends.

This can be addressed in a different PR, as the issue was already there. Let me know if we should create a separate issue for that

https://user-images.githubusercontent.com/54408225/226567569-51f03efb-7f0d-4086-95e7-07184cd252cd.mp4

**Comment by bschorchit** [comment]
Agree with Jyoti on creating a separate issue for this :)
Nice catch on this detail!

**Comment by seaona** [comment]
Thank you both @jpuri @bschorchit ! Issue created [here](https://github.com/MetaMask/metamask-mobile/issues/6008)

**Comment by bschorchit** [comment]
Should this PR be merged? As the reported issues are being handled in the other PR or in the new created ticket? @seaona

**Comment by seaona** [comment]
Looks good!!
Note some QA worked from this PR happened [here](https://github.com/MetaMask/metamask-mobile/pull/5887)

**Pull request merged**